src/com/aca/homework/week4/snake/Snake.java

```diff
@@ -0,0 +1,146 @@
+package com.aca.homework.week4.snake;
+
+import java.util.Random;
+import java.util.Scanner;
+
+public class Snake {
+    private char[][] board;
+    private int[][] snake;
+    private int snakeSize = 1;
+    private int appleRow;
+    private int appleCol;
+    private boolean running = true; // if the player loses, the variable becomes false
+
+    public Snake(char[][] board, int headRow, int headCol, int appleRow, int appleCol) {
+        this.board = board;
+        this.snake = new int[board.length*board[0].length][2];
+        this.snake[snakeSize-1][0] = headRow;
+        this.snake[snakeSize-1][1] = headCol;
+        this.board[headRow][headCol] = '*';
+        this.appleRow = appleRow;
+        this.appleCol = appleCol;
+        this.board[appleRow][appleCol] = '=';
+    }
+    public String getMove(){
+        Scanner scanner = new Scanner(System.in);
+        String move = scanner.nextLine();
+        return move;
+    }
+    public void move(){
+        String move = this.getMove();
+        int dirX = 0, dirY = 0;
+        if("w".equals(move)){
+            dirY = -1;
+        }else if("s".equals(move)){
+            dirY = 1;
+        }else if("a".equals(move)){
+            dirX = -1;
+        }else if("d".equals(move)){
+            dirX = 1;
+        }
+
+
+        if(this.snakeSize > 1 && this.snake[0][0]+dirY == this.snake[1][0] && this.snake[0][1]+dirX == this.snake[1][1]){
+            return;
+        }
+        if(this.isOutOfBounds(snake[0][0]+dirY, snake[0][1]+dirX)){
+            System.out.println("You lose");
+            this.running = false;
+            return;
+        }
+        if(snake[0][0]+dirY == this.appleRow && this.snake[0][1]+dirX == this.appleCol){
+            this.updateSnake(dirY, dirX);
+            this.updateBoard();
+            this.generateApple();
+            this.show();
+            this.snakeSize++;
+        }
+
+        else if (this.snake[0][0] == this.appleRow && this.snake[0][1] == this.appleCol){ // generate a new apple
```

Comment:
remember, every time you write a comment like '// generate a new apple' you failed. you could move the logic to a method with a proper name so we could read the method name and understood that it will generate a new apple.